core/src/main/java/org/jdbi/v3/core/statement/MessageFormatTemplateEngine.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jdbi.v3.core.statement;
+
+import java.text.MessageFormat;
+import java.util.AbstractMap;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.Set;
+
+/**
+ * Uses the equivalent of {@link MessageFormat#format(String, Object...)} as a template engine.
+ *
+ * You must use "0", "1", "2", etc as keys: start at 0, increment by 1. Keys must be numerically unique. You must {@link org.jdbi.v3.core.config.Configurable#define(String, Object)} as many key/value pairs as there are placeholders in the pattern string.
+ *
+ * You may {@code define} key/value pairs in any order. Keys may contain leading {@code '0'}s.
+ *
+ * Any invalid use will trigger an {@link IllegalArgumentException} (or subclasses such as {@link NumberFormatException}) when {@link #render(String, StatementContext)} is called – typically when the statement is about to be executed.
+ *
+ * Example usage:
+ * <pre>{@code
+ *     // select bar from foo where col = 'abc'
+ *     jdbi.useHandle(handle -> handle.createCall("select {1} from {0} where col = ''{2}''")
```

Comment:
Do the `}` in the message format markers end the `@code` block prematurely?

GitHub code highlighting suggests they do.

Reply:
Looks alright to me
![image](https://user-images.githubusercontent.com/3950300/38631480-29d4b8d0-3dba-11e8-8ab5-6448ece69832.png)